.github/workflows/main.yml

```diff
@@ -0,0 +1,425 @@
+        if: matrix.coverage == 'codecov'
+        run: |
+          luacov
+          bash /tmp/codecov-bash -f build/luacov.report.out -X gcov -X coveragepy -F luacov
+
+      - name: Run themes tests
+        # TODO: This fails for older LGI versions.
+        if: matrix.lgi_version == ''
+        run: cd "${{ github.workspace }}/build" && make check-themes
+
+      - name: Upload code coverage report
+        if: matrix.coverage == 'codecov'
+        run: |
+          luacov
+          bash /tmp/codecov-bash -f build/luacov.report.out -X gcov -X coveragepy -F luacov
+
+      - name: Upload awesome binary coverage report
+        if: matrix.coverage == 'codecov'
+        run: |
+          # Report coverage for each .gcno file separately.
+          # gcov will create .gcov files for the same source (e.g. for
+          # globalconf.h from awesome.c.gcno and event.c.gcno).
+          i=0
+          cd "${{ github.workspace }}/build"
+          find -path "*/lgi-check.dir" -prune -o \( -name '*.gcno' -print \) | while read -r gcno; do
+            gcov -pb "$gcno"
+
+            mkdir gcov.$(( ++i ))
+            mv *.gcov "gcov.$i"
+
+            # Delete any files for /usr.
+            # They are not relevant and might cause "Invalid path part" errors
+            # with Code Climate.
+            find "gcov.$i" -maxdepth 1 -type f -name '#usr#*.gcov' -delete
+          done
+
+          # Upload to Codecov.
+          bash /tmp/codecov-bash -X gcov -X coveragepy -F gcov
+
+      - name: Merge coverage
+        if: matrix.coverage == 'coveralls'
+        # Coveralls doesn't support GitHub Actions with `service_name` + `service_job_id` (yet?),
+        # but `luacov-coveralls` behaves as if they did by default.
+        # We need to explicitly remove the `service_name`, so that Coveralls uses the
+        # `repo_token` instead.
+        # See "Referencing a repository": https://docs.coveralls.io/api-introduction
+        run: |
+          luacov-coveralls \
+            --verbose \
+            --merge \
+            --repo-token ${{ secrets.COVERALLS_REPO_TOKEN }} \
+            --service-name ""
+
+      # `check-qa` is the only test that doesn't get a coverage report, so it has to run after all of that.
+      - name: Run qa tests
+        if: matrix.check_qa
+        run: cd "${{ github.workspace }}/build" && make check-qa
+
+      # Check each commit separately (to make git-bisect less annoying).
+      - name: Test previous commits
+        if: matrix.test_prev_commits && github.event_name == 'pull_request'
+        run: |
+          # `actions/checkout` creates a shallow repo (`--depth 1`) by default,
+          # which is fine for everything up until now. But we need individual commits now.
+          # And we only want to unshallow now, to not slow down the checkout for other jobs.
+          git fetch --unshallow
+
+          rev_list="$(git rev-list --bisect-all origin/${{ github.base_ref }}..origin/${{ github.head_ref }})"
```

Comment:
True. It may have only been working for me because I was opening PRs within my own repo. But the docs are quite lackluster on what these values can be.
However, they wouldn't be there if it was as simple as `HEAD`.

My best bet would be to simply drop the `origin/` and hope that `base_ref` and `head_ref` get adjusted according to the context the workflow is run in. This might be another thing that can only be tested after this has been merged.

Reply:
Hmm I guess we better get it merged and see afterwards if things actually work. I feel that dropping `origin` part there is the correct approach, but let's see..